### PR TITLE
fix: inbound share should return only from share endpoint

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

1. fix: inbound share should return only from share endpoint
2. add test cases about query share data from same tenant.

Closes #issue

### CODE REVIEW

**Review by drmingdrmer — APPROVED** [review]

**Comment by BohuTANG** [comment]
```
[ FAIL ] - result differs with:
--- /runner/_work/databend/databend/tests/suites/3_stateful_sharing/01_sharing/01_0000_basic_sharing.result	2023-04-11 11:08:32.529290584 +0000
+++ /runner/_work/databend/databend/tests/suites/3_stateful_sharing/01_sharing/01_0000_basic_sharing.stdout	2023-04-11 11:16:57.708466465 +0000
@@ -1,17 +1,12 @@
 drop test share and database
 prepare shared database and table
-INBOUND test_share test_database shared_tenant shared_tenant
 OUTBOUND test_share test_database shared_tenant shared_tenant,to_tenant
 get shared database and table from same tenant
 INBOUND test_share test_database shared_tenant shared_tenant
 INBOUND test_share test_database shared_tenant shared_tenant
 OUTBOUND test_share test_database shared_tenant shared_tenant,to_tenant
-1
-2
-3
-4
-5
-6
+ERROR 1105 (HY000) at line 1: Code: 1111, Text = Empty share config for creating operator of shared table test_share.t1(while in query flight).
+ERROR 1105 (HY000) at line 1: Code: 1111, Text = Empty share config for creating operator of shared table test_share.t2(while in query flight).
 get shared database and table from another tenant
 INBOUND test_share test_database to_tenant to_tenant
 INBOUND test_share test_database shared_tenant to_tenant
@@ -40,9 +35,7 @@
 4
 5
 6
-4
-5
-6
+ERROR 1105 (HY000) at line 1: Code: 1111, Text = Empty share config for creating operator of shared table test_share.t2(while in query flight).
 ERROR 1105 (HY000) at line 1: Code: 1025, Text = error: 
   --> SQL:1:15
   |
```

File: scripts/ci/deploy/databend-query-sharing.sh [hunk 1]
```diff
@@ -31,3 +31,12 @@ python3 scripts/ci/wait_tcp.py --timeout 10 --port 23003
 echo "Start query node2 for sharding data"
 nohup target/${BUILD_PROFILE}/databend-query -c scripts/ci/deploy/config/databend-query-node-share-2.toml &
 python3 scripts/ci/wait_tcp.py --timeout 30 --port 13307
+
+echo 'Start query node3 open-sharing...'
+nohup target/${BUILD_PROFILE}/open-sharing --tenant=shared_tenant --storage-type=s3 --storage-s3-bucket=testbucket --storage-s3-endpoint-url=http://127.0.0.1:9900 --storage-s3-access-key-id=minioadmin --storage-s3-secret-access-key=minioadmin --storage-allow-insecure --share-endpoint-address=127.0.0.1:23103 &
+python3 scripts/ci/wait_tcp.py --timeout 10 --port 23103
+
+echo "Start query node3 with the same tenant and meta server for sharding data"
+nohup target/${BUILD_PROFILE}/databend-query -c scripts/ci/deploy/config/databend-query-node-share-3.toml &
```
Comment (Xuanwo) [inline scripts/ci/deploy/databend-query-sharing.sh:40]: What will happen if a request is sent to query node 1 to access the sharing table?

Comment (lichuang) [reply scripts/ci/deploy/databend-query-sharing.sh:40]: Now cannot get `inbound` shared tables without `open-sharding` service

Comment (Xuanwo) [reply scripts/ci/deploy/databend-query-sharing.sh:40]: I'm a bit confusing about the config value. Do we need to keep config value of `sharing_endpoint_address` the same across the same cluster?

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**